### DISCUSSION TITLE
Stripe Integration

### DISCUSSION THREAD

**Discussion by jsayles** [nadineproject/nadine#463] [Q&A]
@proffalken I'm breaking this conversation into it's own question so it doesn't get lost in the other thread.  I'm still working out how to use GH Discussions so I'm not sure the most graceful way of doing that is.  

As for Stripe, let's take a step back.  The word "Integration" gets over used quite a bit and is too vague to really indicate what functionality it is trying to achieve.  There are multiple points where a stripe integration makes sense:  Prompting a member to add a card, regular billing/running the cards, managing expired cards, etc. 

The first one is just a hook.  You can use the hooks that already exist or create your own.  All the hooks can be connected to Zapier if you want to use that service to hook things up.  From there you need to set that user up on a recurring bill.  This is best done in stripe so that stripe is the one regularly billing customers.   If the member's bill is the same each month then this is sufficient.  You only need to make changes if the member leaves or their card expires.  This again can be set up using the existing hooks OR you can simply use the Member Alerts that would prompt you to go in and adjust their membership in stripe.  

Perhaps I should have lead with this but Nadine does not automate payments... taking people's money... It simply manages the "This person owes this much money".  The actual payments are done elsewhere.... in this case stripe.  For Office Nomads we use our own merchant service account since the rates are much better (1.8% vs 2.9).

**Comment by proffalken** [comment]
Thanks, that makes sense, and perhaps I should have been clearer on my original comments.

Adding back in the "missing" lines sets up the billing details withing Nadine, however those details don't seem to appear in Stripe to enable the recurring billing even though it was the stripe form that popped up to enter the details.

I'm worried that somewhere I've missed another line that needs to be uncommented so that the user is registered in Stripe.

I'll find more time to work on it in the next day or so, but any tips you can give in order to get the registration in Stripe working would be great!

**Reply by jsayles** [reply]
More than anything you are running into the fact that the stripe implementation was never used anywhere in production so it needs some love.  

Another thing we need to work on is an integration framework that allows for modules to be added/removed.  The reason I commented out the stripe stuff is that it's presence was breaking other things in code I DO use every day.  Same for the elocky and rocketchat code added by the French space.  I talked with them about engineering a better modular system but we never ended up doing it.

**Reply by proffalken** [reply]
Ok, cool, makes sense, I'll spend more time in the code and see what I can find.

Modular approach sounds good.

**Reply by proffalken** [reply]
@jsayles - Looks like Stripe has had a complete overhaul of how to use their system since the original code went in:

https://stripe.com/docs/payments/checkout/migration?lang=python#client-subscriptions

I'll try and find time to port it over - I assume you'd be open to setting the currency and currency symbols as variables as well to enable localisation?

**Comment by proffalken** [comment]
Hey, just to say I've not forgotten about this, I'm hoping to look at it over the Christmas break if I get some time!
